futures-util/src/stream/stream/peek.rs

```diff
@@ -67,21 +70,26 @@ impl<St: Stream> Peekable<St> {
     ///
     /// This method polls the underlying stream and return either a reference
     /// to the next item if the stream is ready or passes through any errors.
-    pub fn poll_peek(
+    pub fn peek(self: Pin<&mut Self>) -> Peek<'_, St> {
+        Peek { inner: Some(self) }
+    }
+
+    fn poll_peek(
         mut self: Pin<&mut Self>,
         cx: &mut Context<'_>,
-    ) -> Poll<Option<&St::Item>> {
+    ) -> Either<Pin<&mut Self>, Poll<Option<&St::Item>>> {
```

Comment:
Given that this never returns `Either::Right(Poll::Pending)` I think changing it to `Either<Pin<&mut Self>, Option<&St::Item>>` would make it a little more readable.

Reply:
(Maybe with a comment that `Left` is equivalent to `Pending` and `Right` is equivalent to `Ready`).

Reply:
Ah, I see.